lib/Pool.js

```diff
@@ -264,16 +272,33 @@ Pool.prototype._removeWorker = function(worker) {
  *                                  after finishing all tasks currently in
  *                                  progress. If true, the workers will be
  *                                  terminated immediately.
+ * @param {number} [timeout]        If provided and non-zero, worker termination promise will be rejected
+ *                                  after timeout if worker process has not been terminated.
+ * @return {Promise.<void, Error>}
  */
-// TODO: rename clear to terminate
-Pool.prototype.clear = function (force) {
-  this.workers.forEach(function (worker) {
-    // TODO: implement callbacks when a worker is actually terminated, only then clear the worker from our array
-    //       else we get zombie child processes :)
-    worker.terminate(force);
+Pool.prototype.terminate = function (force, timeout) {
+  var f = function (worker) {
+    this._removeWorkerFromList(worker);
+  };
+  const removeWorker = f.bind(this);
+
+  const promises = [];
+  const workers = this.workers.slice();
+  workers.forEach(function (worker) {
+    const termPromise = worker.terminateAndNotify(force, timeout)
+      .then(removeWorker);
+    promises.push(termPromise);
   });
+  return Promise.all(promises).then();
```

Comment:
Is there a special reason to append this line with `.then()` ? That doesn't do anything. (same on line 301)

Reply:
No, not really. It got rid of a warning in WebStorm that I wasn't handling the promise, which can of course just be ignored. The only argument I can make for it, which is somewhat weak, is that it clearly signals our intention to ignore the promise. I am fine with dropping that.
I decide to leave the 'clear' call in most of the tests because it really isn't necessary in the tests to change to using a promise. That little WebStorm warning showed up in all the tests without that, and it just annoyed me. 🙂